### PR TITLE
Filter and sort event stream

### PR DESCRIPTION
**Pull request by breedx-splk**
JFR does not guarantee that events are time sorted.  In order to detect that a thread for a given stack frame was working as part of a span (and to link the two) we need to sort.

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/FilterSortedRecordingFile.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.stream.Stream;
+import jdk.jfr.consumer.RecordedEvent;
+
+/**
+ * Limits a delegate stream to just the events that we care about and returns them sorted by start
+ * time.
+ */
+public class FilterSortedRecordingFile implements RecordedEventStream {
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/FilterSortedRecordingFile.java:29]:
```suggestion
class FilterSortedRecordingFile implements RecordedEventStream {
```

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/FilterSortedRecordingFile.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  private final RecordedEventStream.Factory delegateStreamFactory;
+
+  public FilterSortedRecordingFile(RecordedEventStream.Factory delegateStreamFactory) {
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/FilterSortedRecordingFile.java:33]:
```suggestion
FilterSortedRecordingFile(RecordedEventStream.Factory delegateStreamFactory) {
```

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.splunk.opentelemetry.profiler.events;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class RelevantEvents {
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java:24]:
```suggestion
class RelevantEvents {
```

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  public static Set<String> EVENT_NAMES =
```
Comment (jkwatson) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java:25]:
```suggestion
  static Set<String> EVENT_NAMES =
```

Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java:25]: It is used out side the package.

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java (hunk 3 above)
Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java:24]: It's used outside the package.

Comment (jkwatson) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/events/RelevantEvents.java:24]: Ah, I had missed the sub-package here. sorry for the noise

**Pull request merged**